src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java

```diff
@@ -207,7 +209,8 @@ public String description() {
      * executed. The filter {@code operation} is not expected to handle the
      * exchange or {@linkplain HttpExchange#sendResponseHeaders(int, long) send the response headers}.
      * Doing so is likely to fail, since the exchange has commonly been handled
-     * before the operation is invoked.
+     * before the {@code operation} is invoked. Correspondingly, the client
+     * commonly receives the response before the filter operation is executed.
```

Comment:
I'd rather replace `commonly` with `may`: this is essentially racy and you cannot assume that any of the outcome is more likely than the other.

```
     * [...]  Correspondingly, the client
     * may receive the response before the filter operation is executed.

Reply:
Maybe rephrase? Who knows what the client will received, if anything - since the bytes will be sent over the network.

Maybe something like:
  More specifically, the response will typically be sent before the filter operation is executed.